src/config/proxy.ts

```diff
@@ -1,15 +1,27 @@
 import { envVars }  from "./env";
-import { bootstrap } from "global-agent";
-import { getLogger } from "./logger";
-
-const logger = getLogger("config.proxy");
-
-if (envVars.PROXY) {
-	logger.info(`configuring proxy: ${envVars.PROXY} for outbound calls`);
-	process.env.GLOBAL_AGENT_HTTP_PROXY = envVars.PROXY;
-	bootstrap();
-} else {
-	logger.info("configuring no proxy for outbound calls");
-}
+import { AxiosRequestConfig } from "axios";
+import { HttpsProxyAgent } from "https-proxy-agent";
+
+const outboundProxyHttpsAgent = envVars.PROXY ? new HttpsProxyAgent(envVars.PROXY) : undefined;
+
+/**
+ * Use this Axios config to make HTTP(S) requests via the configured outbound proxy.
+ */
+export const OUTBOUND_PROXY_CONFIG: Partial<AxiosRequestConfig> = {
+	// Even though Axios provides the `proxy` option to configure a proxy, this doesn't work and will
+	// always cause an HTTP 501 (see https://github.com/axios/axios/issues/3459). The workaround is to
+	// create an HttpsProxyAgent and set the `proxy` option to false.
+	httpsAgent: outboundProxyHttpsAgent,
+	proxy: false
+};
```

Comment:
This Axios issue drove me nuts. Why is there a `proxy` option to configure a proxy host and port when it ALWAYS returns a 501?

Reply:
apparently, it has something to do with a combination of sub package node minimum TLS versions. This might be fixed in the latest version of Axios (0.27.2).  Maybe do an update while you're at it?